### ISSUE TITLE
JSONParser.parse not working correctly for nested Integers with a magnitude greater than 2^24

### ISSUE THREAD

**Issue by deldrid1** [electricimp/JSONParser#3]
Take the following example:

``` squirrel
local t =[ time()]
server.log(http.jsonencode(t))
server.log(JSONEncoder.encode(t))

server.log(http.jsondecode(http.jsonencode(t))[0])
server.log(JSONParser.parse(JSONEncoder.encode(t))[0])

server.log("////////////////////")
local t =time()
server.log(http.jsonencode(t))
server.log(JSONEncoder.encode(t))

server.log(http.jsondecode(http.jsonencode(t)))
server.log(JSONParser.parse(JSONEncoder.encode(t)))

server.log("\\\\\\\\\\\\\\\\\\\\\\")
local t ={ "t": time() }
server.log(http.jsonencode(t))
server.log(JSONEncoder.encode(t))

server.log(http.jsondecode(http.jsonencode(t)).t)
server.log(JSONParser.parse(JSONEncoder.encode(t)).t)
```

When run on an agent, it logs the following

```
2016-05-04 13:48:44 UTC-4   [Agent] [ 1462384124 ]
2016-05-04 13:48:44 UTC-4   [Agent] [1462384124]
2016-05-04 13:48:44 UTC-4   [Agent] 1462384124
2016-05-04 13:48:44 UTC-4   [Agent] 1.46238e+09     //<- This shouldn't be in scientific notation! 
2016-05-04 13:48:44 UTC-4   [Agent] ////////////////////
2016-05-04 13:48:44 UTC-4   [Agent] 1462384124
2016-05-04 13:48:44 UTC-4   [Agent] 1462384124
2016-05-04 13:48:44 UTC-4   [Agent] 1462384124
2016-05-04 13:48:44 UTC-4   [Agent] 1462384124
2016-05-04 13:48:44 UTC-4   [Agent] \\\\\\\\\\\
2016-05-04 13:48:44 UTC-4   [Agent] { "t": 1462384124 }
2016-05-04 13:48:44 UTC-4   [Agent] {"t":1462384124}
2016-05-04 13:48:44 UTC-4   [Agent] 1462384124
2016-05-04 13:48:44 UTC-4   [Agent] 1.46238e+09   //<- loss of precision here too
```

I've not stepped through the code yet but it appears that nested ints get parsed as floats, even though they might be ints.  The issue is that for any value above 2^24=16,777,216 (which time() is a good candidate for) there will be a loss of precision and unexpected things will happen...

**Comment by terrencebarr** [comment]
Thanks, will look into this

**Comment by betzrhodes** [comment]
Number types now parsed as both integers and floats, so big integers behave correctly.  Fix is in v1.0.0 which has now been released as an official library.
